Services/OnScreenChat/classes/Provider/OnScreenChatNotificationProvider.php

```diff
@@ -0,0 +1,129 @@
+                ->withNewAmount(0)
+            );
+
+        return [
+            $group,
+        ];
+    }
+
+    /**
+     * Delivers async a new item defined by the data sent through GET
+     */
+    public function getAsyncItem(){
+        $name = $_GET["name"];
+        $message = $_GET["message"];
+
+        //Define some Endpoint of the close action
+        $async_url = $this->dic->ctrl()->getLinkTargetByClass('ilOnScreenChatGUI','asyncCloseItem');
+
+        // Note, I am unsere about the state of the user avatars, therefore I would just use the
+        // Chat Icon for the moment.
+        $chat_icon = $this->dic->ui()->factory()->symbol()->icon()->standard("chtr","conversations");
+        $chat_title = $this->dic->ui()->factory()->link()->standard("Conversations", '#');
+        $chat_notification_item = $this->dic->ui()->factory()->item()->notification($chat_title,$chat_icon)
+                                            ->withDescription("No Conversations available");
+
+        if($_GET["no_aggregates"] !== "true"){
+            //Create the aggregate (sub) items
+            $aggregate_title = $this->dic->ui()->factory()->button()->shy($name, '#')
+                                         ->withAdditionalOnLoadCode(function($id){
+                                             //Do what needs to be done by clicking on the title probably reload the entries without the one here.
+                                             return "$('#$id').click(function() {
+                    return console.log('Notification Title has been clicked');
+                });";
+                                         });
+            $aggregate_item = $this->dic->ui()->factory()->item()->notification($aggregate_title,$chat_icon)
+                                        ->withDescription($message ."1")
+                                        ->withCloseAction($async_url."&nr=1");
+
+            $chat_notification_item = $chat_notification_item
+                ->withAggregateNotifications([$aggregate_item])
+                ->withDescription("You have 1 conversation");
+        }
+
+        echo $this->dic->ui()->renderer()->renderAsync([$chat_notification_item]);
+        exit;
+    }
+
+    /**
+     * Allows to perform work after closing the Icon on server and client.
+     */
+    public function asyncCloseItem(){
+        $nr = $_GET['nr'];
+        $async_url = $this->dic->ctrl()->getLinkTargetByClass('ilOnScreenChatGUI','getNewAsyncItem');
+        $async_url .= "&no_aggregates=true";
+        $js = $this->dic->ui()->factory()->legacy("")->withOnLoadCode(function($id) use ($nr,$async_url){
+            //Do what neeeds to be done, if the icon is closed
+            return "
+               il.ChatScopeNotificationItem.replaceByAsyncItem('$async_url');
+               console.log('Notification nr $nr has been closed');
+            ";
+        });
+        echo $this->dic->ui()->renderer()->renderAsync($js);
+        exit;
```

Comment:
Hi @Amstutz 
I'm not a fan of exiting in these classes, is there another way ( e.g. by returning something that is defined and print that out and exiting from the outside?)

Reply:
Nobody is a fan of those cases ;-). If you give me a pattern to change those, I gladly change.